packages/dds/map/src/directory.ts

```diff
@@ -1551,7 +1558,12 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
         // This should make the subdirectory structure unreachable so it can be GC'd and won't appear in snapshots
         // Might want to consider cleaning out the structure more exhaustively though?
         const successfullyRemoved = this._subdirectories.delete(subdirName);
-        this.disposeSubDirectoryTree(previousValue);
+        if (previousValue !== undefined) {
+            this.disposeSubDirectoryTree(previousValue);
+            const absolutePath = posix.join(this.absolutePath, subdirName);
+            this.directory.emit("subDirectoryDeleted", absolutePath, local, this.directory);
```

Comment:
Same here, prefer to watch from the `SharedDirectory` and echo the event.

Reply:
Done